src/rpc/services/isForkDisputedService/IsForkDisputedService.ts

```diff
@@ -0,0 +1,106 @@
+import { ARpcService } from "@/rpc";
+import { ChannelId, ForkId, Timestamp } from "@/types/types";
+import ATransport from "@/transport/ATransport";
+import P2PManager from "@/P2PManager";
+import IsForkDisputedRpcMethods from "./IsForkDisputedRpcMethods";
+
+class IsForkDisputedService extends ARpcService<IsForkDisputedRpcMethods> {
+    // Track acknowledged disputed forks
+    acknowledgedDisputedForks: WeakMap<ATransport, Set<ForkId>> = new WeakMap();
+
+    constructor(p2pManager: P2PManager) {
+        super(p2pManager);
+    }
+
+    public createRPCMethods(transport: ATransport): IsForkDisputedRpcMethods {
+        return new IsForkDisputedRpcMethods(transport, this);
+    }
+
+    /**
+     * Request all peers to acknowledge a disputed fork
+     * This should be called when a dispute window is created on-chain
+     */
+    public requestDisputeAcknowledgment(channelId: ChannelId, forkId: ForkId) {
+        console.log(
+            `Requesting all peers to acknowledge disputed fork ${forkId}`
+        );
+
+        // Create a snapshot of openConnections at the time of request
+        // This captures the peers we sent the request to, so we don't disconnect
+        // from peers that connect after we sent the request
+        const snapshotTransports = [...this.p2pManager.openConnections];
+
+        // Broadcast the request
+        this.remoteRpc.isForkDisputedService
+            .onDisputeAcknowledgmentRequest(channelId, forkId)
+            .broadcast();
+
+        setTimeout(
```

Comment:
we have a timeoutManager now - best we use that for all setTimeouts, since it takes into account disposal and potentially canceling tasks